src/pynwb/__init__.py

```diff
@@ -297,7 +303,11 @@ def read(self, **kwargs):
                 raise TypeError("NWB version %s not supported. PyNWB supports NWB files version 2 and above." %
                                 str(file_version_str))
         # read the file
-        return super().read(**kwargs)
+        file = super().read(**kwargs)
+        if self.external_resources is not None:
+            er_read = ExternalResources.from_flat_tsv(path=self.external_resources)
+            file.link_resources(er_read)
```

Comment:
I think the logic for using `external_resources` with the I/O should probably move to HDMF,  probably at the level of `HDMFIO`. We will need the same logic also in `NWBZarrIO` and the logic should be the same for `HDF5IO` and `ZarrIO`. Similarly, `external_resources` should then be a parameter on `HDMFIO.__init__`. @rly what do you think?

Reply:
I'll create a ticket on hdmf and move the I/O logic there.